### PR TITLE
Extract communication volume

### PR DESCRIPTION
**Pull request by jmferreiratech**
This set of patches aim to separate the concept of edge weight from graph theory and that of link weight from our domain. I extracted out the high level concept thus letting the low level one free to be used to more 'static' things as the link faut probability. Now Edges are finally immutable and the LinkWeight will depend on the set of paths passed to it.

### CODE REVIEW

**Comment by jmferreiratech** [comment]
@rmilfont and @rafaelgmota, could you please review my changes request?

File: src/StatisticalAnalyser.java [hunk 1]
```diff
@@ -65,53 +65,64 @@ public double averageLinkWeight(ArrayList<ArrayList<Path>> paths) {
...
             for(Path path: alp)
-                accumulatedPathWeight += path.getWeight()/(double)(path.size()-1);
+                accumulatedPathWeight += lwTracker.weight(path)/(double)(path.size()-1);
         }
         return accumulatedPathWeight/(double)nPaths;
     }
 
+    private LinkWeightTracker linkWeightTracker(ArrayList<ArrayList<Path>> paths) {
+        LinkWeightTracker lwTracker = new LinkWeightTracker(graph);
+        for (ArrayList<Path> samePairPaths : paths) {
```
Comment (rafaelgmota) [inline src/StatisticalAnalyser.java:118]: Why this logic isn't a method of LinkWeightTracker class?

Comment (jmferreiratech) [reply src/StatisticalAnalyser.java:118]: LinkWeightTracker isn't aware of the separation of paths by communication pair.

Comment (rmilfont) [reply src/StatisticalAnalyser.java:118]: The constructor of LinkWeightTracker need to receive the communication volume. Also, the constructor of StatisticalAnalyser also needs this change.

**Review by rmilfont — CHANGES_REQUESTED** [review]

**Pull request merged**